be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/MessageDecoder.scala

```diff
@@ -89,6 +89,7 @@ object MessageDecoder {
       case (ObjectType.ELECTION, ActionType.END) => request.toTypedRequest(JsonRpcRequestEndElection)
       case (ObjectType.MESSAGE, ActionType.WITNESS) => request.toTypedRequest(JsonRpcRequestWitnessMessage)
       case (ObjectType.CHIRP, ActionType.ADD) => request.toTypedRequest(JsonRpcRequestAddChirp)
+      case (ObjectType.CHIRP, ActionType.DELETE) => request.toTypedRequest(JsonRpcRequestDeleteChirp)
```

Comment:
I think we should also have add_broadcast and delete_broadcast here. Otherwise you'll get an "invalid object/action pair" (which is not really the case).

I guess you didn't put it because the server is not supposed to get these message. If so, what I would do instead is add two new cases for add_broadcast and delete_broadcast and return a PipelineError such as "Server does not support yet ...". The reason I would do that is that when we will have consensus on be2, the server may have to check/treat these messages